### PR TITLE
chore: add test for new client generation

### PR DESCRIPTION
**Pull request by meltsufin**
Fixes: #8659

### CODE REVIEW

**Comment by suztomo** [comment]
> because the repository is private.

How about yoshi-code-bot account? https://github.com/googleapis/google-cloud-java/blob/main/.github/workflows/readme.yaml#L35 (I don't know its entitlement)

**Comment by meltsufin** [comment]
> > because the repository is private.
> 
> How about yoshi-code-bot account? https://github.com/googleapis/google-cloud-java/blob/main/.github/workflows/readme.yaml#L35 (I don't know its entitlement)

Doesn't seem to be working either.

File: .github/workflows/newclient.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      - name: Push to branch
+        run: |
+          [ -z "`git config user.email`" ] && git config --global user.email "${USERNAME:-script}@google.com"
+          [ -z "`git config user.name`" ] && git config --global user.name "${USERNAME:-script}"
+          git checkout -b newclient_output_${{ github.event_name }}
+          git add --all
+          git commit -m 'feat: re-generated apikeys'
+          git remote add monorepo https://${{ github.actor }}:${{ github.token }}@github.com/${{ github.repository }}.git
+          git fetch -q --unshallow monorepo
+          git push -f monorepo newclient_output_${{ github.event_name }}
```
Comment (suztomo) [inline .github/workflows/newclient.yaml:58]: How does "Push to branch" help verify the behavior of new-client.py?

File: .github/workflows/newclient.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      - name: Test
+        run: mvn test -T C1 -B
```
Comment (suztomo) [inline .github/workflows/newclient.yaml:60]: Can you add source code comment that "java-apikeys is included at the aggregator pom.xml"?

File: .github/workflows/newclient.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          java-version: 11
+      - run: java -version
+      - uses: actions/cache@v3
+        id: mvn-cache
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-unified-${{ steps.date.outputs.date }}
+      - name: Install dependencies
+        run: pip install -r generation/new_client/requirements.txt
+      - name: Generate
```
Comment (suztomo) [inline .github/workflows/newclient.yaml:40]: Can you add source code comment (or name) "regenerating existing apikeys module"?

File: .github/workflows/newclient.yaml (hunk 2 above)
Comment (suztomo) [reply .github/workflows/newclient.yaml:60]: How about "-f java-apikeys/pom.xml" to skip irrelevant modules?

File: .github/workflows/newclient.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        run: |
+          python generation/new_client/new-client.py generate \
```
Comment (suztomo) [inline .github/workflows/newclient.yaml:42]: Can you explicitly remove java-apikeys module before new-client.py?

**Comment by suztomo** [comment]
How might we verify new-client.py’s logic of adding a corresponding new entry to gapic BOM, release please configuration files, and the root pom.xml?

File: .github/workflows/newclient.yaml (hunk 1 above)
Comment (meltsufin) [reply .github/workflows/newclient.yaml:58]: It's just there for debugging, in case the job fails.

Comment (suztomo) [reply .github/workflows/newclient.yaml:58]: That's helpful. Nice.

File: .github/workflows/newclient.yaml (hunk 3 above)
Comment (meltsufin) [reply .github/workflows/newclient.yaml:40]: Yep

File: .github/workflows/newclient.yaml (hunk 4 above)
Comment (meltsufin) [reply .github/workflows/newclient.yaml:42]: I do that in new-client.py. Is that not enough?

**Review by suztomo — APPROVED** [review]

File: .github/workflows/newclient.yaml (hunk 4 above)
Comment (suztomo) [reply .github/workflows/newclient.yaml:55]: For me it's itchy that if, by any chahce, new-client.py does nothing, the verification in this check still passes.

Comment (suztomo) [reply .github/workflows/newclient.yaml:55]: It's done below:

```
      - name: Remove Client
        run: |
          rm -rf java-${API_SHORT_NAME} || true
          sed -i "/${API_SHORT_NAME}/d" pom.xml
          sed -i "/${API_SHORT_NAME}/d" gapic-libraries-bom/pom.xml
``

**Pull request merged**